chapter04/ex_4_4.pl

```diff
@@ -0,0 +1,22 @@
+#!perl
+use 5.010;
+
+&greet( "Fred" );
+&greet( "Barney" );
+&greet( "Wilma" );
+&greet( "Betty" );
+
+sub greet {
+    state $last;
+
+    my $name = shift @_;
+
+    if( defined $last ) {
+        printf "Hi %-10s! %s is also here!\n", $name, $last;
```

Comment:
%10s will print in 10 spaces.
%-10s will align it left. So the output will give a tab feeling to read it better(I think).